lib/ThreeScaleClient.php

```diff
@@ -94,7 +107,7 @@ public function getHost() {
    */
   
      public function authorize($appId, $appKey = null, $credentials_or_service_id, $usage = null) {
-    $url = "http://" . $this->getHost() . "/transactions/authorize.xml";
+    $url = "http://" . $this->getHost() . ":" . $this->getPort() . "/transactions/authorize.xml";
```

Comment:
Setting the port without setting the scheme is not that useful. I'd consider setting the endpoint as a whole instead of scheme + host + port .